flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/util/LazyMemorySegmentPool.java

```diff
@@ -37,37 +34,26 @@
     private static final long PER_REQUEST_MEMORY_SIZE = 16 * 1024 * 1024;
 
     private final Object owner;
-    private final @Nullable MemoryManager memoryManager;
+    private final MemoryManager memoryManager;
     private final ArrayList<MemorySegment> cachePages;
     private final int maxPages;
     private final int perRequestPages;
 
     private int pageUsage;
 
-    public LazyMemorySegmentPool(Object owner, MemoryManager memoryManager, long memorySize) {
-        this(
-                owner,
-                memoryManager,
-                (int) memorySize
-                        / (memoryManager == null
-                                ? MemoryManager.DEFAULT_PAGE_SIZE
-                                : memoryManager.getPageSize()));
-    }
-
     public LazyMemorySegmentPool(Object owner, MemoryManager memoryManager, int maxPages) {
```

Comment:
The nullable `MemoryManager` is introduced this month, see facd51fcf5c78c5f733c203c38e3a093ba8a5218. This is introduced because streaming operator can't use MemoryManager at that time, and we have to allocate memory from JVM heap. But with FLINK-20860 is resolved, we don't need this hack code anymore. 

 `BufferDataOverWindowOperator` uses `getContainingTask().getEnvironment().getMemoryManager()` to get the `MemoryManager` which should never be null in our cases, becuase we will never use `DummyEnvironment` in our tests.